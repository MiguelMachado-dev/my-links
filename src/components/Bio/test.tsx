import { describe, it, expect } from 'vitest'
import { render, screen } from '@testing-library/react'
import { Bio } from '.'

const bioData = {
  title: 'OlÃ¡ ðŸ‘‹',
  content:
    'Tenho 24 anos e trabalho como Frontend Engineer na Uber. Entusiasta de tecnologia e amante de jogos!',
}

describe('<Bio />', () => {
  it('should render <Bio /> correctly', () => {
    render(<Bio data={bioData} />)

    const heading = screen.getByRole('heading', {
      name: /olÃ¡ ðŸ‘‹/i,
    })

    const content = screen.getByText(
      /tenho 24 anos e trabalho como frontend engineer na uber\. entusiasta de tecnologia e amante de jogos!/i
    )

    expect(heading).toBeInTheDocument()
    expect(content).toBeInTheDocument()

    expect(heading).toMatchSnapshot()
    expect(content).toMatchSnapshot()
  })
})
