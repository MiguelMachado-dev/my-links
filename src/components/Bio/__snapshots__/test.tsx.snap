// Vitest Snapshot v1

exports[`<Bio /> > should render <Bio /> correctly 1`] = `
.c0 {
  font-size: 2.4rem;
  font-weight: 700;
  line-height: 1.5;
  padding-bottom: 1.6rem;
}

<h3
  class="c0"
>
  OlÃ¡ ðŸ‘‹
</h3>
`;

exports[`<Bio /> > should render <Bio /> correctly 2`] = `
.c0 {
  font-size: 1.6rem;
  line-height: 1.5;
  font-weight: 500;
}

<p
  class="c0"
>
  Tenho 24 anos e trabalho como Frontend Engineer na Uber. Entusiasta de tecnologia e amante de jogos!
</p>
`;
